### DISCUSSION TITLE
How can OpenTelemetry be integrated into a C project? I saw that there is no C among the supported languages, only C++. Is there any case of integrating it into a C project?

### DISCUSSION THREAD

**Discussion by niuzhi** [open-telemetry/opentelemetry-cpp#2340] [Q&A]
How can OpenTelemetry be integrated into a C project? I saw that there is no C among the supported languages, only C++. Is there any case of integrating it into a C project?
https://opentelemetry.io/docs/instrumentation/

**Comment by lalitb** [comment]
OpenTelemetry does not currently provide an official API for the C language. To integrate OpenTelemetry into a C project, one would need to develop a C-compatible wrapper using the available OpenTelemetry C++ or Rust SDKs. As of now, to the best of our knowledge, there isn't any publicly known, dedicated C bridge for OpenTelemetry based on these SDKs.

**Comment by malkia** [comment]
The closest I've seen to a "tracing" api for "C" is Sentry's native - https://github.com/getsentry/sentry-native/blob/master/include/sentry.h#L1687 - but it's not OTEL., it's their own format.

**Comment by k0st1a** [comment]
Is the repository https://github.com/dorsal-lab/opentelemetry-c good enough?
